### PR TITLE
[Emulator tests] Add proxy server - 3

### PR DESCRIPTION
**Pull request by Tulsishah**
### Description
- Follow up PR on https://github.com/GoogleCloudPlatform/gcsfuse/pull/2735
- Added main.go file to start proxy server on host 8020.
- Added debug flag to turn on debug logs.

### Link to the issue in case of a bug fix.
NA

### Testing details
1. Manual - NA
2. Unit tests - NA
3. Integration tests - NA

### CODE REVIEW

**Review by BrennaEpp — COMMENTED** [review]
A couple nits

File: tools/integration_tests/emulator_tests/proxy_server/main.go [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+	"os/signal"
+	"syscall"
+	"time"
+)
+
+var (
+	// Flag to accept config-file path.
+	fConfigPath = flag.String("config-path", "configs/config.yaml", "Path to the file")
+	// Flag to turn on debug logs.
+	debug = flag.Bool("debug", false, "logs will be enabled with a flag value of true.")
```
Comment (raj-prince) [inline tools/integration_tests/emulator_tests/proxy_server/main.go:35]: Can we add `f` prefix in the flag variable?

File: tools/integration_tests/emulator_tests/proxy_server/main.go [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	for name, values := range r.Header {
+		for _, value := range values {
+			req.Header.Add(name, value)
+		}
+	}
+
+	// TODO: Handler request and add retry-id according to the instruction.
```
Comment (raj-prince) [inline tools/integration_tests/emulator_tests/proxy_server/main.go:61]: Do we need this TODO?

File: tools/integration_tests/emulator_tests/proxy_server/main.go [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+
+// Host address of the proxy server.
+// TODO: Allow this value to be configured via a command-line flag.
+const PORT = "8020"
+
+type ProxyHandler struct {
+	http.Handler
+}
+
+func (ph ProxyHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
```
Comment (raj-prince) [inline tools/integration_tests/emulator_tests/proxy_server/main.go:48]: Can we write a high level doc - what does it do?

File: tools/integration_tests/emulator_tests/proxy_server/main.go [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+	// Send the request to the target server
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	respURL, err := resp.Location()
+	// Change the resp URL host(target host) to proxy server host.
```
Comment (raj-prince) [inline tools/integration_tests/emulator_tests/proxy_server/main.go:72]: Can we write `why` is this required?

File: tools/integration_tests/emulator_tests/proxy_server/main.go [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+
+	// Copy the response body
+	w.WriteHeader(resp.StatusCode)
+	_, err = io.Copy(w, resp.Body)
+	if err != nil {
+		log.Printf("Error in coping response body: %v", err)
+	}
+}
+
+// ProxyServer represents a simple proxy server
```
Comment (raj-prince) [inline tools/integration_tests/emulator_tests/proxy_server/main.go:102]: // ProxyServer represents a simple proxy server over GCS storage based API endpoint.

File: tools/integration_tests/emulator_tests/proxy_server/main.go (hunk 1 above)
Comment (raj-prince) [reply tools/integration_tests/emulator_tests/proxy_server/main.go:35]: Also, we can keep a todo to support log-file path later.

**Review by raj-prince — COMMENTED** [review]
Looks good, few minor comments.

File: tools/integration_tests/emulator_tests/proxy_server/main.go (hunk 2 above)
Comment (Tulsishah) [reply tools/integration_tests/emulator_tests/proxy_server/main.go]: I will add request handler in upcoming PR to add ID in request header, that's why I added TODO.

File: tools/integration_tests/emulator_tests/proxy_server/operation_manager.go [hunk 6]
```diff
@@ -32,7 +32,11 @@ func NewOperationManager(config Config) *OperationManager {
 	for _, retryConfig := range config.RetryConfig {
 		om.addRetryConfig(retryConfig)
 	}
-	log.Printf("%+v\n", om)
+
+	if *debug {
+		log.Printf("%+v\n", om)
```
[the remark this answers is not in the document]
Comment (Tulsishah) [reply tools/integration_tests/emulator_tests/proxy_server/operation_manager.go]: Makes sense. As we don't need time stamp and all to debug this we can use println directly. Updated

**Review by raj-prince — APPROVED** [review]

**Pull request merged**